core/src/main/scala/cats/Inject.scala

```diff
@@ -0,0 +1,54 @@
+package cats
+
+/**
+ * Inject is a type class to inject a value of type `A` into a
+ * coproduct of types `B` when `A :≺: B`. The coproduct `B` is
+ * modelled with `Either`.
```

Comment:
My understanding of the `Inject` type class is that it represents an injective function between type `A` and `B`.  If we treat types as sets, and a function between the two types are just pairs of elements in the two sets `A` and `B`. Each pair has an "input" element in set `A` and an "output" element of `B`.  For an injective function, for every element in `A` there is exactly one pair. And for every element in `B` there is either one pair or no pair at all. 

A total pure scala function is an injective function, but it only specifies pairs for all elements in `A`, i.e. given any `A` you can get a `B`. It doesn't specify pairs for elements in `B`, this `Inject` type class does that through `proj`.